src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/RecepBeginningOfMessage.java

```diff
@@ -0,0 +1,47 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+
+@Getter
+@Setter
+@RequiredArgsConstructor
+public class RecepBeginningOfMessage extends Segment {
```

Comment:
We could consider modifying BeginningOfMessage to handle these parameters (currently is assumes they are all empty) - but I don't think we need to here is why:

I don't think there is any information of value for us in this segment. There is no information in BGM that we would ever need to translate into a FHIR message or record in a database. We just have to make sure we can write it correctly into the outbound edifact files.

Reply:
I'm not sure I understand what code change you are proposing here

Reply:
I am also not sure what code change I am proposing here.